packages/core/src/components/document-component/document-component-heading.tsx

```diff
@@ -1,24 +1,44 @@
 import { h, FunctionalComponent } from "@stencil/core";
 import { JSXBase } from "@stencil/core/internal";
+import { DocumentComponentMode } from "./document-component.models";
 
 interface DocumentComponentHeadingProps {
   heading: "h2" | "h3" | "h4" | "h5" | "h6";
+  mode: DocumentComponentMode;
+  href?: string;
 }
 
 export const Heading: FunctionalComponent<
   DocumentComponentHeadingProps & JSXBase.HTMLAttributes<HTMLHeadingElement>
 > = ({ heading, ...props }, children) => {
+  let headingElement: HTMLHeadingElement;
   switch (heading) {
     default:
     case "h2":
-      return <h2 {...props}>{children}</h2>;
+      headingElement = <h2 {...props}>{children}</h2>;
+      break;
     case "h3":
-      return <h3 {...props}>{children}</h3>;
+      headingElement = <h3 {...props}>{children}</h3>;
+      break;
     case "h4":
-      return <h4 {...props}>{children}</h4>;
+      headingElement = <h4 {...props}>{children}</h4>;
+      break;
     case "h5":
-      return <h5 {...props}>{children}</h5>;
+      headingElement = <h5 {...props}>{children}</h5>;
+      break;
     case "h6":
-      return <h6 {...props}>{children}</h6>;
+      headingElement = <h6 {...props}>{children}</h6>;
+  }
+
+  if (props.mode === "document") {
+    return headingElement;
+  }
+
+  if (props.mode === "table-of-contents" && props.href) {
+    return (
+      <a href={props.href} class="heading-anchor">
+        {headingElement}
+      </a>
+    );
```

Comment:
Wat als er geen `props.href` is meegegeven?

Reply:
Tja, dan rendert er niks. ... Als een afnemer het Document Component met mode="table-of-contents" inzet zal er ook een href mee gegeven moet worden. Wat stel jij voor dan?

Reply:
Ik zou het gedrag van `props.mode === "document"` als fallback/default gedrag nemen:

```ts
  if (props.mode === "table-of-contents" && props.href) {
    return (
      <a href={props.href} class="heading-anchor">
        {headingElement}
      </a>
    );
  }

  return headingElement;
```

Dan doet het tenminste "iets" 🤷. Zomaar `undefined` uit de `render()` teruggeven is wellicht iets te kort door de bocht.

Reply:
Ok. Heb ik gedaan.

Reply:
Ah, grappig dat je regels 33-35 hebt laten staan 😄

Reply:
Die zijn nu weg 😶‍🌫️

Reply:
💨